src/Castle.Windsor/MicroKernel/DefaultKernel.cs

```diff
@@ -537,6 +537,10 @@ public IKernel Register(params IRegistration[] registrations)
 			var token = OptimizeDependencyResolution();
 			foreach (var registration in registrations)
 			{
+				if (registration is FromAssemblyDescriptor)
```

Comment:
Sorry for not getting to review this sooner.

Is there a reason you put this in the kernel rather than in the descriptor itself like I suggested in the issue?

https://github.com/castleproject/Windsor/blob/c192ed68f3b8ea22f0cd67f3922c9029a5de58e2/src/Castle.Windsor/MicroKernel/Registration/FromDescriptor.cs#L173-L178

If you didn't want the exception to also apply to `FromTypesDescriptor` (which has the same type of problem), then I think it would be best to override `Register` in `FromAssemblyDescriptor` and perform the check there before calling `base`.